www/class/centreonAuth.SSO.class.php

```diff
@@ -194,91 +224,126 @@ protected function checkSsoClient()
                 }
             }
         } else {
-            # Only SSO (no login from local users)
+            // Only SSO (no login from local users)
             return 1;
         }
     }
 
     protected function checkPassword($password, $token = "", $autoimport = false)
     {
         if ($this->ssoMandatory == 1) {
-            # Mode LDAP autoimport. Need to call it
+            // Mode LDAP autoimport. Need to call it
             if ($autoimport) {
-                # Password is only because it needs one...
+                // Password is only because it needs one...
                 parent::checkPassword('test', $token, $autoimport);
             }
-            # We delete old sessions with same SID
+            // We delete old sessions with same SID
             global $pearDB;
             $pearDB->query("DELETE FROM session WHERE session_id = '" . session_id() . "'");
             $this->passwdOk = 1;
         } else {
-            # local connect (when sso not enabled and 'sso_mode' == 1
+            // local connect (when sso not enabled and 'sso_mode' == 1
             return parent::checkPassword($password, $token);
         }
     }
 
 
     /**
-     * Connect to Keycloak and get token access
+     * Connect to OpenId Connect and get token access
      *
-     * @param string $base Keycloak Server Url
-     * @param string $realm Keycloak Client Realm
-     * @param string $redirectUri Keycloak Redirect Url
-     * @param string $clientId Keycloak Client ID
-     * @param string $clientSecret Keycloak Client Secret
-     * @param string $code Keycloak Authorization Code
+     * @param string $url          OpenId Connect Client Token endpoint
+     * @param string $redirectUri  OpenId Connect Redirect Url
+     * @param string $clientId     OpenId Connect Client ID
+     * @param string $clientSecret OpenId Connect Client Secret
+     * @param string $code         OpenId Connect Authorization Code
+     * @param bool   $verifyPeer   disable SSL verify peer
      *
      * @return string
     */
-    public function getKeycloakToken($base, $realm, $redirectUri, $clientId, $clientSecret, $code)
+    public function getOpenIdConnectToken($url, $redirectUri, $clientId, $clientSecret, $code, $verifyPeer)
     {
-        $url = $base . "/realms/" . $realm . "/protocol/openid-connect/token";
-        $data = array(
+        $data = [
             "client_id" => $clientId,
             "client_secret" => $clientSecret,
             "grant_type" => "authorization_code",
             "code" => $code,
             "redirect_uri" => $redirectUri
-        );
+        ];
+
         $ch = curl_init($url);
```

Comment:
could you add a try catch and execute the curl_close in the finally

Reply:
cURL does not throw exceptions, it is a thin wrapper for the C library it binds to.